snap/plugins/juju_go.py

```diff
@@ -22,137 +22,99 @@
-        schema = super().schema()
-        schema["properties"]["go-channel"] = {
-            "type": "string",
-            "default": "latest/stable",
-        }
-        schema["properties"]["go-packages"] = {
-            "type": "array",
-            "minitems": 1,
-            "uniqueItems": True,
-            "items": {"type": "string"},
-            "default": [],
-        }
-        schema["properties"]["go-external-strings"] = {
+    def get_schema(cls) -> Dict[str, Any]:
+        return {
+            "$schema": "http://json-schema.org/draft-04/schema#",
             "type": "object",
-            "additionalProperties": {"type": "string"},
-            "default": {},
-        }
-        schema["properties"]["go-static"] = {
-            "type": "boolean",
-            "default": False,
+            "additionalProperties": False,
+            "properties": {
+                "go-channel": {"type": "string", "default": "latest/stable"},
+                "go-buildtags": {
+                    "type": "array",
+                    "uniqueItems": True,
+                    "items": {"type": "string"},
+                    "default": [],
+                },
+                "go-packages": {
+                    "type": "array",
+                    "minitems": 1,
+                    "uniqueItems": True,
+                    "items": {"type": "string"},
+                    "default": ["./..."],
+                },
+                "go-external-strings": {
+                    "type": "object",
+                    "additionalProperties": {"type": "string"},
+                    "default": {},
+                },
+                "go-static": {
+                    "type": "boolean",
+                    "default": False,
+                },
+                "go-strip": {
+                    "type": "boolean",
+                    "default": False,
+                },
+            },
+            "required": ["source"],
         }
-        schema["properties"]["go-strip"] = {
-            "type": "boolean",
-            "default": False,
+
+    def get_build_snaps(self) -> Set[str]:
+        return {f"go/{self.options.go_channel}"}
+
+    def get_build_packages(self) -> Set[str]:
+        return {"gcc"}
+
+    def get_build_environment(self) -> Dict[str, str]:
+        env = {
+            "GOBIN": "${SNAPCRAFT_PART_INSTALL}/bin",
         }
-        return schema
+        if self.options.go_static:
+            env.update({
+                "CGO_ENABLED": "0"
+            })
 
-    @classmethod
-    def get_build_properties(cls):
-        # Inform Snapcraft of the properties associated with building. If these
-        # change in the YAML Snapcraft will consider the build step dirty.
-        return ["go-packages", "go-external-strings", "go-channel", "go-static", "go-strip"]
-
-    def __init__(self, name, options, project):
-        super().__init__(name, options, project)
-        self.build_packages.extend(["git"])
```

Comment:
If and when we ever need to include packages directly from git, we can add this back in then. As long as we trust the offical go package proxy, we should be fine without it.